### PR TITLE
feat: Obfuscate shards

### PR DESCRIPTION
**Pull request by jan-gogo**
Related to #1818

Unit tests pass.

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: corellium/shard/obfuscate/src/main/kotlin/flank/corellium/shard/Obfuscate.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package flank.corellium.shard
+
+/**
+ * Obfuscate each [Shard.Test.Case.name] value.
+ * Use for security purpose.
+ *
+ * @receiver calculated shards
+ * @return obfuscated shards
+ */
+fun Shards.obfuscate(): Shards =
```
Comment (pawelpasterz) [inline corellium/shard/obfuscate/src/main/kotlin/flank/corellium/shard/Obfuscate.kt:10]: Hm... this code is quite hard to read and might be difficult to maintain in the future

Comment (jan-gogo) [reply corellium/shard/obfuscate/src/main/kotlin/flank/corellium/shard/Obfuscate.kt:10]: I agree that our lint rules are annoying. The first version was:
```
fun Shards.obfuscate(): Shards =
    map { shard ->
        shard.map { app ->
            app.copy(tests = app.tests.map { test ->
                test.copy(cases = test.cases.map { case ->
                    case.copy(name = obfuscationMappings.obfuscateAndroidTestName(case.name))
                })
            })
        }
    }
```

@pawelpasterz Any propositions?

Comment (pawelpasterz) [reply corellium/shard/obfuscate/src/main/kotlin/flank/corellium/shard/Obfuscate.kt:10]: TBH I never thought about linter. What I meant - code in this form may be error prone. I'm aware it's partially enforced by structure of `Shard` (and that is fine).
That's why it's just a comment, not a change request -- I point there might be a problem here in the future, but don't have better solution on my mind

Comment (jan-gogo) [reply corellium/shard/obfuscate/src/main/kotlin/flank/corellium/shard/Obfuscate.kt]: Ok. I have added explicit types and some comments that explain the meaning of this function.

**Review by pawelpasterz — APPROVED** [review]

**Review by Sloox — APPROVED** [review]

**Pull request merged**